### PR TITLE
Module refactoring and deprecation cleanup

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Module refactoring and deprecation cleanup.  The list of module renames are:
    - Rename `Cardano.CLI.Parsers` to `Cardano.CLI.Options`
    - Rename `Cardano.CLI.Ping` to `Cardano.CLI.Run.Ping`
    - Rename `Cardano.CLI.Shelley.Output` to `Cardano.CLI.Types.Output`
    - Rename `Cardano.CLI.Shelley.Key` to `Cardano.CLI.Types.Key`
    - Rename `Cardano.CLI.Shelley.Orphans` to `Cardano.CLI.Orphans`
    - Rename `Cardano.CLI.Conway.Types` to `Cardano.CLI.Types.Governance`
    - Rename `Cardano.CLI.Types` to `Cardano.CLI.Types.Legacy`
    - Rename `Cardano.CLI.Common.Parsers` to `Cardano.CLI.Options.Common`
    - Rename `Cardano.CLI.Shelley.Parsers` to `Cardano.CLI.Options.Legacy`
    - Rename `Cardano.CLI.Conway.Parsers` to `Cardano.CLI.Options.Governance`
    - Rename `Cardano.CLI.Conway.Commands` to `Cardano.CLI.Commands.Governance`
    - Rename `Cardano.CLI.Shelley.Commands.*` to `Cardano.CLI.Commands.Legacy.*`
    - Rename `Cardano.CLI.Shelley.Run.*` to `Cardano.CLI.Run.Legacy.*`
    - Rename `Cardano.CLI.Shelley.Run` to `Cardano.CLI.Run.Legacy`
    - Rename `*ShelleyClient*` to `*LegacyClient*, *ShelleyCommand*` to `*LegacyCommand*`
    - Rename `Cardano.CLI.Run.Friendly` to `Cardano.CLI.Json.Friendly`
  compatibility: compatible
  type: maintenance
```

# Context

Rename some modules in preparation for new top-level command groups.  In particular remove mention of `Conway` and `Shelley` from module names.  These can be confusing when top-level era command groups are introduced.  Anything mentioning `Conway` will use `Governance` instead and anything mention `Shelley` will use `Legacy`. 

The idea with `Legacy` is to have a namespace to hold code and we can move code out of it when we find a better location for them.  For example `Certificate` or `Governance` or `Transaction` submodule perhaps.  And when `Legacy` is emptied out, we can remove it.

`Shelley` is gone.  The `Shelley` module was really `ShelleyBased` as it contained code relevant to future eras as well.  This PR removes it because shelley-based is assume.

The PR also introduce a consume of module grouping under the following modules:

* `Cardano.CLI.Commands` - Command types and support functions
* `Cardano.CLI.Options` - CLI options parsers
* `Cardano.CLI.Run` - Run functions for commands
* `Cardano.CLI.Types` - CLI types and supporting functions

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Environment.hs [hunk 1]
```diff
@@ -55,3 +59,23 @@ getEnvNetworkId = do
 -- Otherwise, return 'Nothing'.
 getEnvSocketPath :: IO (Maybe FilePath)
 getEnvSocketPath = IO.lookupEnv "CARDANO_NODE_SOCKET_PATH"
+
+-- | If the environment variable @CARDANO_ERA@ is set, then return the set value.
+-- Otherwise, return 'Nothing'.
+getCardanoEra :: IO (Maybe AnyCardanoEra)
+getCardanoEra = do
+  mEraString <- IO.lookupEnv "CARDANO_NODE_SOCKET_PATH"
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Environment.hs]: Should this be `"CARDANO_ERA"`?

File: cardano-cli/src/Cardano/CLI/OS/Unix.hs [hunk 2]
```diff
@@ -1,6 +1,9 @@
 {-# LANGUAGE CPP #-}
 
-module Cardano.CLI.OS.Unix where
+module Cardano.CLI.OS.Unix
+  ( module System.Posix.Files
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/OS/Unix.hs]: I don't think this will compile

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Environment.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/Environment.hs]: Fixed.  Thanks!

File: cardano-cli/src/Cardano/CLI/OS/Unix.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/OS/Unix.hs]: Fixed!

**Pull request merged**